### PR TITLE
[WIP] Added ByEquivalentTo by assert action

### PR DESCRIPTION
**Pull request by DanielIndie95**
## IMPORTANT 
SINCE FLUENT ASSERTIONS IS CURRENTLY UNDER HEAVY REFACTORING FOR [VERSION 5.0](https://github.com/fluentassertions/fluentassertions/issues/463), WE CURRENTLY ONLY ACCEPT BUGFIX REQUESTS. SORRY FOR THE INCONVENIENCE

* [ ] The [Pull Request](https://help.github.com/articles/using-pull-requests) is targeted at the `master` branch.
* [ ] The code complies with the [Coding Guidelines for C# 3.0, 4.0 and 5.0](http://www.csharpcodingguidelines.com/)/. 
* [ ] The changes are covered by a new or existing set of unit tests which follow the Arrange-Act-Assert syntax such as is used [in this example](https://github.com/fluentassertions/fluentassertions/blob/daaf35b9b59b622c96d0c034e8972a020b2bee55/Tests/FluentAssertions.Shared.Specs/BasicEquivalencySpecs.cs#L33). 
* [ ] If the contribution affects the documentation, please help us by providing a paragraph of text and/or examples that we can include in the [documentation](https://github.com/fluentassertions/fluentassertions/wiki) wiki.

### CODE REVIEW

**Comment by DanielIndie95** [comment]
didnt quite get why the tests failed (failed before running the tests)

**Comment by dennisdoomen** [comment]
Thank you for this contribution, but I'll have to postpone merging this PR until we've completed the internal reorganization of the code-based for FA 5.0.

File: Tests/FluentAssertions.Shared.Specs/GenericCollectionAssertionsSpecs.cs [hunk 1]
```diff
@@ -34,6 +34,444 @@ public void When_asserting_equality_with_a_collection_built_from_params_argument
...
+
+        [TestMethod]
+        public void Check_asserting_equivallency_by_assert_method_should_check_it_regardless_of_the_order()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            //Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            List<DateTime> actualDates = new List<DateTime>
+            {
+                DateTime.Today,
```
Comment (krajek) [inline Tests/FluentAssertions.Shared.Specs/GenericCollectionAssertionsSpecs.cs]: Please use constant values in tests. It seems that this test can fail on the edge of the day.
In general, using non-deterministic data in tests is a code smell.

File: Tests/FluentAssertions.Shared.Specs/GenericCollectionAssertionsSpecs.cs [hunk 2]
```diff
@@ -34,6 +34,444 @@ public void When_asserting_equality_with_a_collection_built_from_params_argument
...
+            //-----------------------------------------------------------------------------------------------------------
+            //Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            const string BECAUSE = "there are some missing members";
+            List<DateTime> missingItems = new List<DateTime>
+            {
+                DateTime.Today.AddDays(2),
+                DateTime.Today.AddDays(7)
+            };
+            String missingItemsMessage = String.Join(", ", missingItems.Select(it => "<" + it.ToString("yyyy-MM-dd") + ">"));
```
Comment (krajek) [inline Tests/FluentAssertions.Shared.Specs/GenericCollectionAssertionsSpecs.cs]: Do not duplicate library logic in tests. Please use constant values for actual and expected values and manually prepare expected message

File: Src/Core/Collections/GenericCollectionAssertions.cs [hunk 3]
```diff
@@ -18,164 +18,143 @@ public GenericCollectionAssertions(IEnumerable<T> actualValue) : base(actualValu
         }
 
         /// <summary>
-        /// Asserts that a collection is ordered in ascending order according to the value of the specified
+        /// Asserts that a collection is ordered in ascending order according to the value of the the specified
```
Comment (krajek) [inline Src/Core/Collections/GenericCollectionAssertions.cs]: Duplicated 'the'

File: Src/Core/Collections/GenericCollectionAssertions.cs [hunk 4]
```diff
@@ -18,164 +18,143 @@ public GenericCollectionAssertions(IEnumerable<T> actualValue) : base(actualValu
...
-            Expression<Func<T, TSelector>> propertyExpression, IComparer<TSelector> comparer, string because = "", params object[] args)
+            Expression<Func<T, object>> propertyExpression, string because = "", params object[] args)
         {
-            return BeOrderedBy(propertyExpression, comparer, SortOrder.Ascending, because, args);
+            return BeOrderedBy(propertyExpression, SortDirection.Ascending, because, args);
         }
 
         /// <summary>
-        /// Asserts that a collection is ordered in descending order according to the value of the specified
+        /// Asserts that a collection is ordered in descending order according to the value of the the specified
```
Comment (krajek) [inline Src/Core/Collections/GenericCollectionAssertions.cs]: Duplicated 'the', here and in few other places

File: Tests/FluentAssertions.Shared.Specs/GenericCollectionAssertionsSpecs.cs [hunk 5]
```diff
@@ -34,6 +34,444 @@ public void When_asserting_equality_with_a_collection_built_from_params_argument
...
+            assert.ShouldThrow<AssertFailedException>();
+        }
+
+        [TestMethod]
+        public void Check_asserting_equivallency_by_assert_method_should_throw_exception_when_context_collection_is_bigger()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            //Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            List<DateTime> actualDates = new List<DateTime>
```
Comment (krajek) [inline Tests/FluentAssertions.Shared.Specs/GenericCollectionAssertionsSpecs.cs]: Please replace some tests for datetimes with other types(ints, strings, custom classes and so on). Also, please provide tests for collections with null values.

**Comment by dennisdoomen** [comment]
@DanielIndie95 do you still want this in? The `release-5.0` branch is now in a state that allows you to safely rebase it on.

**Comment by dennisdoomen** [comment]
@DanielIndie95 can you please let us know what your intentions are with this?

**Comment by DanielIndie95** [comment]
@dennisdoomen i still want it , but after the refactor the code have changed so i need to change my code as well. Ill try to fix it in the next few days

**Comment by dennisdoomen** [comment]
No activity for the last 6 months.
